File: example/javalib/web/3-todo-spring-boot/src/com/example/todoitem/TodoItem.java
package com.example.todoitem;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.validation.constraints.NotBlank;

@Entity
public class TodoItem {
  @Id
  @GeneratedValue
  private Long id;

  @NotBlank
  private String title;

  private boolean completed;

  protected TodoItem() {}

  public TodoItem(String title, boolean completed) {
    this.title = title;
    this.completed = completed;
  }

  public Long getId() {
    return id;
  }

  public String getTitle() {
    return title;
  }

  public void setTitle(String title) {
    this.title = title;
  }

  public boolean isCompleted() {
    return completed;
  }

  public void setCompleted(boolean completed) {
    this.completed = completed;
  }
}

File: example/kotlinlib/web/8-todo-spring-boot/src/com/example/todoitem/TodoItem.kt
package com.example.todoitem

import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.validation.constraints.NotBlank

@Entity
data class TodoItem(
    @Id
    @GeneratedValue
    val id: Long? = null,

    @field:NotBlank
    var title: String = "",

    var completed: Boolean = false,
)
